internal/pam/pam_test.go

```diff
@@ -40,9 +40,10 @@ func TestAuthenticate(t *testing.T) {
 		"error on unexisting conf":                            {conf: "doesnotexist.conf", wantErrType: pam.ErrPamSystem},
 		"error on unexisting users":                           {username: "no such user", wantErrType: pam.ErrPamAuth},
 		"error on invalid password":                           {username: "invalid credentials", wantErrType: pam.ErrPamAuth},
-		"error on offline with user online user not in cache": {conf: "forceoffline.conf", initialCache: "db_with_old_users", wantErrType: pam.ErrPamAuth},
-		"error on offline with purged user account":           {username: "veryolduser@domain.com", initialCache: "db_with_old_users", wantErrType: pam.ErrPamAuth},
-		"error on offline with unpurged old user account":     {conf: "forceoffline-expire-right-away.conf", initialCache: "db_with_old_users", username: "veryolduser@domain.com", wantErrType: pam.ErrPamAuth},
+		"error on offline with user online user not in cache": {conf: "forceoffline.conf", initialCache: "db_with_expired_users", wantErrType: pam.ErrPamAuth},
+		"error on offline with purged user account":           {username: "purgeduser@domain.com", initialCache: "db_with_expired_users", wantErrType: pam.ErrPamAuth},
+		"error on offline with unpurged old user account":     {conf: "forceoffline-expire-right-away.conf", initialCache: "db_with_expired_users", username: "purgeduser@domain.com", wantErrType: pam.ErrPamAuth},
+		"error on offline with expired user account":          {conf: "forceoffline.conf", initialCache: "db_with_expired_users", username: "expireduser@domain.com", wantErrType: pam.ErrPamAuth},
```

Comment:
nitpick (because there had to be one :p): in the previous tests you were doing expired, then unpurged, which is the right order IMHO, Please try follow the same order here.